test/cppsuite/test_harness/workload_tracking.h

```diff
@@ -0,0 +1,79 @@
+#ifndef WORKLOAD_TRACKING_H
+#define WORKLOAD_TRACKING_H
+
+#include <iostream>
+
+/* Default schema for tracking table
+ * key_format : Collection name / Key
+ * value_format : Operation type / Value / Time
+ */
+#define DEFAULT_TRACKING_TABLE_SCHEMA "key_format=Si,value_format=iSi"
+
+namespace test_harness {
+/* Tracking operations. */
+enum class tracking_operation { CREATE, INSERT };
+/* Class used to track operations performed on collections */
+class workload_tracking {
+
+    public:
+    workload_tracking(WT_CONNECTION *conn, const std::string &collection_name)
+        : _collection_name(collection_name), _conn(conn), _timestamp(0U)
+    {
+    }
+
+    ~workload_tracking()
+    {
+        if (_session != nullptr) {
+            if (_session->close(_session, NULL) != 0)
+                /* Failing to close session is not blocking. */
+                debug_info(
+                  "Failed to close session, shutting down uncleanly", _trace_level, DEBUG_ERROR);
+            _session = nullptr;
+        }
+    }
+
+    int
+    load(const std::string &table_schema = DEFAULT_TRACKING_TABLE_SCHEMA)
```

Comment:
This should be void.

Reply:
It will be changed to void if needed when the Component interface will be implemented